app/controllers/se_auth_controller.rb

```diff
@@ -46,12 +46,14 @@ def already_done
 
   def deauth
```

Comment:
You've forgotten to remove the `before_action :verify_admin`, so this won't work. In any case, users shouldn't be able to delete the whole `stack_user` record; allow admins to do that, but a user-triggered deauth should just remove the SE API token.

Reply:
If the access token is removed, how are users going to regenerate a new one?

Reply:
If they're de-authorizing the app, why would they _want_ to generate a new one?

Reply:
Because they wanted to or accidentally invalidated one?

Reply:
I don't see where you're going. The deauth button is there to remove details from the app if someone doesn't want to use it any more. If that's the case, there's no need to be able to regenerate a token.

Reply:
Well, in most of the apps/userscripts I use, users always have the option to deauthorize and then authorize again. OTOH, you may be right - should I revert the commit?

Reply:
Oh, I see what you're aiming at. No, this is good as-is in that case.